test/integration/clustercidr/ipam_test.go

```diff
@@ -138,6 +139,456 @@ func TestIPAMMultiCIDRRangeAllocatorType(t *testing.T) {
+			t.Fatal(err)
+		}
+		if gotPodCIDRs, err := nodePodCIDRs(clientSet, node.Name); err != nil {
+			t.Fatal(err)
+		} else if !reflect.DeepEqual(expectedPodCIDRs, gotPodCIDRs) {
+			t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", expectedPodCIDRs, gotPodCIDRs)
+		}
+
+		// Delete the ClusterCIDR
+		if err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Delete(context.TODO(), clusterCIDR.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		//Wait and make sure that the ClusterCIDR is not deleted, as there is a node associated with it.
+		if err := wait.PollImmediate(time.Second, 5*time.Second, func() (bool, error) {
+			cc, err := clientSet.NetworkingV1alpha1().ClusterCIDRs().Get(context.TODO(), clusterCIDR.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			return cc != nil, nil
+		}); err != nil {
+			t.Fatalf("failed while waiting for ClusterCIDR %q to be deleted: %v", clusterCIDR.Name, err)
+		}
+
+		//Delete the node.
+		if err := clientSet.CoreV1().Nodes().Delete(context.TODO(), node.Name, metav1.DeleteOptions{}); err != nil {
+			t.Fatal(err)
+		}
+
+		// Create a node, which should get Pod CIDRs from the ClusterCIDR with fewer matching label Count,
+		// as the best match ClusterCIDR is marked as terminating.
+		node2 := makeNode("dualstack-node-2", map[string]string{"ipv4": "true", "ipv6": "true"})
+		expectedPodCIDRs2 := []string{"10.1.0.0/24", "fd12:30:100::/120"}
+		if _, err := clientSet.CoreV1().Nodes().Create(context.TODO(), node2, metav1.CreateOptions{}); err != nil {
+			t.Fatal(err)
+		}
+		if gotPodCIDRs2, err := nodePodCIDRs(clientSet, node2.Name); err != nil {
+			t.Fatal(err)
+		} else if !reflect.DeepEqual(expectedPodCIDRs2, gotPodCIDRs2) {
+			t.Errorf("unexpected result, expected Pod CIDRs %v but got %v", expectedPodCIDRs2, gotPodCIDRs2)
+		}
+	})
+}
+
+func TestIPAMMultiCIDRRangeAllocatorClusterCIDRTieBreak(t *testing.T) {
+	// set the feature gate to enable MultiCIDRRangeAllocator
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.MultiCIDRRangeAllocator, true)()
+
+	_, kubeConfig, tearDownFn := framework.StartTestServer(t, framework.TestServerSetup{
+		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
+			// Disable ServiceAccount admission plugin as we don't have serviceaccount controller running.
+			opts.Admission.GenericAdmission.DisablePlugins = []string{"ServiceAccount", "TaintNodesByCondition"}
+			opts.APIEnablement.RuntimeConfig.Set("networking.k8s.io/v1alpha1=true")
+		},
+	})
+	defer tearDownFn()
+
+	clientSet := clientset.NewForConfigOrDie(kubeConfig)
+	sharedInformer := informers.NewSharedInformerFactory(clientSet, 1*time.Hour)
+
+	ipamController := booststrapMultiCIDRRangeAllocator(t, clientSet, sharedInformer)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go ipamController.Run(ctx.Done())
+	sharedInformer.Start(ctx.Done())
```

Comment:
I think that you should put all this block inside the t.Run, or you will be sharing state between testcases

Reply:
Yes I tried doing that earlier however it takes much longer to run the test 92.465s vs 25.355s (current implementation). Not sure if it's ok to increase the time required to run the test, also it will keep on increasing if we add more tests.

Thus I added deletes to clean these up:  https://github.com/kubernetes/kubernetes/blob/1e682f1fdab179171b08535eab8f778380596753/test/integration/clustercidr/ipam_test.go#L576-L589

Reply:
that is. good point, then make sure we delete things and start clean